porch/apiserver/pkg/registry/porch/packagerevision.go

```diff
@@ -98,23 +98,17 @@ func (r *packageRevisions) Create(ctx context.Context, runtimeObject runtime.Obj
 		return nil, apierrors.NewBadRequest(fmt.Sprintf("expected PackageRevision object, got %T", runtimeObject))
 	}
 
-	expectedName := obj.Spec.RepositoryName + ":" + obj.Spec.PackageName + ":" + obj.Spec.Revision
-	name := obj.Name
-	if name == "" {
-		name = expectedName
+	if obj.Name != "" {
+		klog.Warningf("Client provided metadata.name %q", obj.Name)
 	}
 
-	if name != expectedName {
-		return nil, apierrors.NewBadRequest(fmt.Sprintf("name should be %q", expectedName))
-	}
-
-	nameTokens, err := ParseName(name)
-	if err != nil {
-		return nil, apierrors.NewBadRequest(fmt.Sprintf("invalid name %q", name))
+	repositoryName := obj.Spec.RepositoryName
```

Comment:
Should we check that spec.repositoryName matches the value encoded name?  Or is the plan to just move away that restriction entirely?

Reply:
For Create, I think we expect it normally to be called with a Name, and I don't think we should default it if not set - I think that would be surprising and would likely break validation.  The functionality is available in GenerateName if we do want it, though it isn't used very often AFAIK

Reply:
will add TODO with the 2nd PR